packages/presentation/lib/ui/splash_screen/bloc/splash_bloc.dart

```diff
@@ -1,33 +1,81 @@
+import 'dart:io';
+
 import 'package:domain/use_case/splash_duration_use_case.dart';
+import 'package:domain/use_case/check_version_use_case.dart';
 import 'package:presentation/base/bloc.dart';
+import 'package:presentation/generated/l10n.dart';
+import 'package:presentation/library/const/links.dart';
+import 'package:presentation/navigation/base_arguments.dart';
 import 'package:presentation/ui/movie_page/movie_widget.dart';
+import 'package:presentation/ui/splash_screen/bloc/splash_tile.dart';
+import 'package:url_launcher/url_launcher.dart';
 
-abstract class SplashBloc extends Bloc {
+abstract class SplashBloc extends Bloc<BaseArguments, SplashTile> {
   factory SplashBloc(
     SplashDurationUseCase durationUseCase,
+    CheckVersionUseCase versionUseCase,
   ) =>
       SplashBlocImpl(
         durationUseCase,
+        versionUseCase,
       );
+
+  Future<void> updateTap();
 }
 
-class SplashBlocImpl extends BlocImpl implements SplashBloc {
+class SplashBlocImpl extends BlocImpl<BaseArguments, SplashTile>
+    implements SplashBloc {
+  var _tile = SplashTile.init();
   final SplashDurationUseCase _durationUseCase;
+  final CheckVersionUseCase _versionUseCase;
 
   SplashBlocImpl(
     this._durationUseCase,
+    this._versionUseCase,
   );
 
   @override
-  void initState() {
+  void initState() async {
     super.initState();
-    nextScreen();
+    final versionResult = await checkVersion();
+    nextScreen(versionResult);
+  }
+
+  Future<String> checkVersion() async {
+    final checkResult = await _versionUseCase();
+    switch (checkResult) {
+      case TypeNotificationVersion.outdatedVersion:
+        _tile = _tile.copyWith(checkResult: S.current.outdatedVersionResult);
+        handleData(tile: _tile);
+        break;
+      case TypeNotificationVersion.suitableVersion:
+        _tile = _tile.copyWith(checkResult: S.current.suitableVersionResult);
+        handleData(tile: _tile);
+        return S.current.suitableVersionResult;
+      case null:
+        handleData(tile: _tile);
+        return S.current.actualVersionResult;
+    }
+    return S.current.emptyString;
   }
 
-  Future<void> nextScreen() async {
+  @override
+  Future<void> updateTap() {
+    if (Platform.isAndroid) {
```

Comment:
move uri to separate variable. it will have checks, yes. but launchUrl is the same for all platforms